File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for input text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int num_letters = count_letters(text);
    int num_words = count_words(text);
    int num_sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    float L = (num_letters / (float) num_words) * 100.0;
    float S = (num_sentences / (float) num_words) * 100.0;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int num_letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            num_letters++;
        }
    }
    return num_letters;
}

int count_words(string text)
{
    int num_words = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            num_words++;
        }
    }
    num_words++;
    return num_words;
}

int count_sentences(string text)
{
    int num_sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            num_sentences++;
        }
    }
    return num_sentences;
}

File: pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Prompt the user for input text
    text = get_string("Text: ")

    # Count the number of letters, words, and sentences in the text
    num_letters = count_letters(text)
    num_words = count_words(text)
    num_sentences = count_sentences(text)

    # Compute the Coleman-Liau index
    L = (num_letters / num_words) * 100
    S = (num_sentences / num_words) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Print the grade level
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    num_letters = 0
    for c in text:
        if c.isalpha():
            num_letters += 1
    return num_letters


def count_words(text):
    num_words = 0
    for c in text:
        if c.isspace():
            num_words += 1
    num_words += 1
    return num_words


def count_sentences(text):
    num_sentences = 0
    for c in text:
        if c == "." or c == "!" or c == "?":
            num_sentences += 1
    return num_sentences


main()
